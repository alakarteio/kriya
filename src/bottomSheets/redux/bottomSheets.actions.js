export const OPEN_BOTTOMSHEETS = 'OPEN_BOTTOMSHEETS'
export const open = (code, contents) => ({ type: OPEN_BOTTOMSHEETS, payload: { code, contents } })

export const CLOSE_BOTTOMSHEETS = 'CLOSE_BOTTOMSHEETS'
export const close = code => ({ type: CLOSE_BOTTOMSHEETS, payload: code })

export const TOGGLE_BOTTOMSHEETS = 'TOGGLE_BOTTOMSHEETS'
export const toggle = code => ({ type: TOGGLE_BOTTOMSHEETS, payload: code })
