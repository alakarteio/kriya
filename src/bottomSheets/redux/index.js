import bottomSheets from './bottomSheets'
import * as actions from './bottomSheets.actions'
import * as selectors from './bottomSheets.selectors'

Object.assign(bottomSheets, actions, selectors)

export * from './bottomSheets.actions'
export * from './bottomSheets.selectors'
export default bottomSheets
