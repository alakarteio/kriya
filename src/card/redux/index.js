export { default } from './card'
export * from './card.actions'
export * from './card.selectors'
