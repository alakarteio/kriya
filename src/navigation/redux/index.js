export { default } from './navigation'
export * from './navigation.actions'
export * from './navigation.selectors'
