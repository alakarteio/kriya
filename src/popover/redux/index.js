export { default } from './popover'
export * from './popover.actions'
export * from './popover.selectors'
