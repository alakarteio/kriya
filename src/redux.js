export { default as cards } from './card/redux'
export { default as navigation } from './navigation/redux'
export { default as popover } from './popover/redux'
export { default as bottomSheets } from './bottomSheets/redux'
export { default as tabs } from './tabs/redux'
