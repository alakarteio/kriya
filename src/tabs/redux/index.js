export { default } from './tabs'
export * from './tabs.actions'
export * from './tabs.selectors'
