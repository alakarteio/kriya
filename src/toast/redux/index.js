import toast from './toast'
import * as actions from './toast.actions'
import * as selectors from './toast.selectors'

Object.assign(toast, actions, selectors)

export * from './toast.actions'
export * from './toast.selectors'
export default toast

