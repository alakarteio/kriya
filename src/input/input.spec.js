/* eslint-disable import/no-extraneous-dependencies, react/jsx-filename-extension */
/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import { reduxForm } from 'redux-form'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import mockComponent from '__mocks__/component'
import Input from './input'
import InputContainer from './input.container'

jest.mock('kriya-select', () => mockComponent('kriya-select'))

jest.mock('./checkbox', () => mockComponent('checkbox'))

const Decorated = reduxForm({ form: 'testForm' })(Input)
const DecoratedContainer = reduxForm({ form: 'formCont' })(InputContainer)

const snap = ({ name = 'name', ...rest }) => {
  const store = createStore(() => ({}))
  const component = renderer.create(
    <Provider store={store}>
      <Decorated {...rest} name={name} />
    </Provider>,
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
}

describe('common/Input', () => {
  describe('graphical (JSX)', () => {
    it('should have a default behaviour', () => snap({}))
    it('should add custom className', () => snap({ className: 'custom' }))
    it('should add custom style', () => snap({ style: { backgroundColor: 'red' } }))

    describe('labels', () => {
      it('should print a label', () => snap({ label: 'my label' }))
      it('should print a * on label when input is required', () => snap({
        required: true,
        label: 'my label',
      }))
    })

    it('should print an icon when this is a select', () => snap({ type: 'select' }))
    it('should print the placeholder', () => snap({ type: 'input', placeholder: 'My placeholder' }))
    it('should be disabled', () => snap({ type: 'input', disabled: true }))

    describe('types', () => {
      it('should print a textarea', () => snap({ type: 'textarea' }))
      it('should print a select', () => snap({ type: 'select' }))
      it('should print a selectbox', () => snap({ type: 'selectbox' }))
      it('should print an input (input)', () => snap({ type: 'input' }))
      it('should print an input (checkbox)', () => snap({ type: 'checkbox' }))
      it('should print an input (radio)', () => snap({ type: 'radio' }))
      it('should print an input (number)', () => snap({ type: 'number' }))

      const options = [
        { value: 'o1', label: 'l1' },
        { value: 'o4', label: 'l4' },
        { value: 'o2', label: 'l2' },
        { value: 'o3', label: 'l3' },
      ]
      it('should print options on select', () => snap({ type: 'select', options }))
      it('should print options on selectbox', () => snap({ type: 'selectbox', options }))
      it('shouldnt print options (not select)', () => snap({ type: 'input', options }))
    })
  })
  describe('container', () => {
    const snapContainer = (state, props) => {
      const store = createStore(() => (
        {
          form: {
            form1: {
              values: {
                field1: '1',
              },
            },
          },
        }
      ))
      const component = renderer.create(
        <Provider store={store}>
          <DecoratedContainer {...props} />
        </Provider>,
      )

      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    }

    it('show placeholder on label when value is not empty', () => snapContainer({}, {
      form: 'form1',
      name: 'field1',
      placeholder: 'placeholder',
      type: 'input',
    }))

    it('hidden label when withoutLabel is true', () => snapContainer({}, {
      form: 'form1',
      name: 'field1',
      withoutLabel: true,
      label: 'label',
      type: 'input',
    }))

    it('show label when withoutLabel is false', () => snapContainer({}, {
      form: 'form1',
      name: 'field1',
      withoutLabel: false,
      label: 'label',
      type: 'input',
    }))

    it('should pass value of redux form for select', () => snapContainer({}, {
      form: 'form1',
      name: 'field1',
      options: [
        { value: 'o1', label: 'l1' },
        { value: 'o4', label: 'l4' },
        { value: 'o2', label: 'l2' },
        { value: 'o3', label: 'l3' },
      ],
      type: 'select',
    }))

    it('should pass value of redux form for selectbox', () => snapContainer({}, {
      form: 'form1',
      name: 'field1',
      options: [
        { value: 'o1', label: 'l1' },
        { value: 'o4', label: 'l4' },
        { value: 'o2', label: 'l2' },
        { value: 'o3', label: 'l3' },
      ],
      type: 'selectbox',
    }))
  })
})

/* eslint-enable import/no-extraneous-dependencies, react/jsx-filename-extension */
