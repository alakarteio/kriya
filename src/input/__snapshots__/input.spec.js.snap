// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/Input container should pass value of redux form for select 1`] = `
<div
  className="input type-select"
  style={Object {}}
>
  <select
    className="field"
    disabled={false}
    name="field1"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    placeholder=""
    required={false}
    type="select"
    value="1"
  >
    <option
      value="o1"
    >
      l1
    </option>
    <option
      value="o4"
    >
      l4
    </option>
    <option
      value="o2"
    >
      l2
    </option>
    <option
      value="o3"
    >
      l3
    </option>
  </select>
  <i
    className="arrow mdv-expand_more"
  />
</div>
`;

exports[`common/Input container should pass value of redux form for selectbox 1`] = `
<div
  className="input type-selectbox"
  style={Object {}}
>
  <div
    className="kriya-select"
  >
    {"name":"kriya-select","props":{"simpleValue":true,"name":"field1","placeholder":"","disabled":false,"required":false,"options":[{"value":"o1","label":"l1"},{"value":"o4","label":"l4"},{"value":"o2","label":"l2"},{"value":"o3","label":"l3"}],"anyTouched":false,"asyncValidating":false,"dirty":true,"form":"form1","initialized":false,"invalid":false,"pristine":false,"submitting":false,"submitFailed":false,"submitSucceeded":false,"valid":true,"pure":true,"array":{},"value":"1"}}
  </div>
</div>
`;

exports[`common/Input graphical (JSX) labels should print a * on label when input is required 1`] = `
<div
  className="input type-input"
  style={Object {}}
>
  <label
    htmlFor="name"
  >
    my label
    *
  </label>
  <input
    className="field"
    disabled={false}
    name="name"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    placeholder=""
    required={true}
    type="input"
    value=""
  />
</div>
`;

exports[`common/Input graphical (JSX) labels should print a label 1`] = `
<div
  className="input type-input"
  style={Object {}}
>
  <label
    htmlFor="name"
  >
    my label
  </label>
  <input
    className="field"
    disabled={false}
    name="name"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    placeholder=""
    required={false}
    type="input"
    value=""
  />
</div>
`;

exports[`common/Input graphical (JSX) should add custom className 1`] = `
<div
  className="input type-input custom"
  style={Object {}}
>
  <input
    className="field"
    disabled={false}
    name="name"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    placeholder=""
    required={false}
    type="input"
    value=""
  />
</div>
`;

exports[`common/Input graphical (JSX) should add custom style 1`] = `
<div
  className="input type-input"
  style={
    Object {
      "backgroundColor": "red",
    }
  }
>
  <input
    className="field"
    disabled={false}
    name="name"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    placeholder=""
    required={false}
    type="input"
    value=""
  />
</div>
`;

exports[`common/Input graphical (JSX) should be disabled 1`] = `
<div
  className="input type-input"
  style={Object {}}
>
  <input
    className="field"
    disabled={true}
    name="name"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    placeholder=""
    required={false}
    type="input"
    value=""
  />
</div>
`;

exports[`common/Input graphical (JSX) should have a default behaviour 1`] = `
<div
  className="input type-input"
  style={Object {}}
>
  <input
    className="field"
    disabled={false}
    name="name"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    placeholder=""
    required={false}
    type="input"
    value=""
  />
</div>
`;

exports[`common/Input graphical (JSX) should print an icon when this is a select 1`] = `
<div
  className="input type-select"
  style={Object {}}
>
  <select
    className="field"
    disabled={false}
    name="name"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    placeholder=""
    required={false}
    type="select"
    value=""
  />
  <i
    className="arrow mdv-expand_more"
  />
</div>
`;

exports[`common/Input graphical (JSX) should print the placeholder 1`] = `
<div
  className="input type-input"
  style={Object {}}
>
  <input
    className="field"
    disabled={false}
    name="name"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    placeholder="My placeholder"
    required={false}
    type="input"
    value=""
  />
</div>
`;

exports[`common/Input graphical (JSX) types should print a select 1`] = `
<div
  className="input type-select"
  style={Object {}}
>
  <select
    className="field"
    disabled={false}
    name="name"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    placeholder=""
    required={false}
    type="select"
    value=""
  />
  <i
    className="arrow mdv-expand_more"
  />
</div>
`;

exports[`common/Input graphical (JSX) types should print a selectbox 1`] = `
<div
  className="input type-selectbox"
  style={Object {}}
>
  <div
    className="kriya-select"
  >
    {"name":"kriya-select","props":{"simpleValue":true,"name":"name","placeholder":"","disabled":false,"required":false,"options":[],"anyTouched":false,"asyncValidating":false,"dirty":false,"form":"testForm","initialized":false,"invalid":false,"pristine":true,"submitting":false,"submitFailed":false,"submitSucceeded":false,"valid":true,"pure":true,"array":{}}}
  </div>
</div>
`;

exports[`common/Input graphical (JSX) types should print a textarea 1`] = `
<div
  className="input type-textarea"
  style={Object {}}
>
  <textarea
    className="field"
    disabled={false}
    name="name"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    placeholder=""
    required={false}
    type="textarea"
    value=""
  />
</div>
`;

exports[`common/Input graphical (JSX) types should print an input (checkbox) 1`] = `
<div
  className="input type-checkbox"
  style={Object {}}
>
  <input
    checked={false}
    className="field"
    disabled={false}
    name="name"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    placeholder=""
    required={false}
    type="checkbox"
    value=""
  />
</div>
`;

exports[`common/Input graphical (JSX) types should print an input (input) 1`] = `
<div
  className="input type-input"
  style={Object {}}
>
  <input
    className="field"
    disabled={false}
    name="name"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    placeholder=""
    required={false}
    type="input"
    value=""
  />
</div>
`;

exports[`common/Input graphical (JSX) types should print an input (number) 1`] = `
<div
  className="input type-number"
  style={Object {}}
>
  <input
    className="field"
    disabled={false}
    name="name"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    placeholder=""
    required={false}
    type="number"
    value=""
  />
</div>
`;

exports[`common/Input graphical (JSX) types should print an input (radio) 1`] = `
<div
  className="input type-radio"
  style={Object {}}
>
  <input
    checked={false}
    className="field"
    disabled={false}
    name="name"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    placeholder=""
    required={false}
    type="radio"
    value={undefined}
  />
</div>
`;

exports[`common/Input graphical (JSX) types should print options on select 1`] = `
<div
  className="input type-select"
  style={Object {}}
>
  <select
    className="field"
    disabled={false}
    name="name"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    placeholder=""
    required={false}
    type="select"
    value=""
  >
    <option
      value="o1"
    >
      l1
    </option>
    <option
      value="o4"
    >
      l4
    </option>
    <option
      value="o2"
    >
      l2
    </option>
    <option
      value="o3"
    >
      l3
    </option>
  </select>
  <i
    className="arrow mdv-expand_more"
  />
</div>
`;

exports[`common/Input graphical (JSX) types should print options on selectbox 1`] = `
<div
  className="input type-selectbox"
  style={Object {}}
>
  <div
    className="kriya-select"
  >
    {"name":"kriya-select","props":{"simpleValue":true,"name":"name","placeholder":"","disabled":false,"required":false,"options":[{"value":"o1","label":"l1"},{"value":"o4","label":"l4"},{"value":"o2","label":"l2"},{"value":"o3","label":"l3"}],"anyTouched":false,"asyncValidating":false,"dirty":false,"form":"testForm","initialized":false,"invalid":false,"pristine":true,"submitting":false,"submitFailed":false,"submitSucceeded":false,"valid":true,"pure":true,"array":{}}}
  </div>
</div>
`;

exports[`common/Input graphical (JSX) types shouldnt print options (not select) 1`] = `
<div
  className="input type-input"
  style={Object {}}
>
  <input
    className="field"
    disabled={false}
    name="name"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    placeholder=""
    required={false}
    type="input"
    value=""
  />
</div>
`;
